src/azure-cli-core/azure/cli/core/aaz/_poller.py

```diff
@@ -12,6 +12,10 @@
 from azure.core.polling.base_polling import LROBasePolling
 from azure.core.tracing.common import with_current_context
 from azure.core.tracing.decorator import distributed_trace
+# import LROPoller in main thread to resolve import deadlock between threads in python 3.10.10
+# reference https://github.com/psf/requests/issues/2925 and https://github.com/Azure/azure-cli/issues/26272
+from msrest.polling.poller import LROPoller  # pylint: disable=unused-import
```

Comment:
My gut feeling tells me this may not be safe enough, as there may be other dependencies imported by the sub-thread and main thread at the same time.

The SDK implementation is safer because the first `PUT` operation is performed on the main thread, so subsequent polling `GET` operations on the sub-thread will have the full dependency tree cached already.

Besides `LROPoller`, there are also `AzureOperationPoller` and `AzureCoreLROPoller` imported from `msrestazure.azure_operation` and `azure.core.polling` respectively:

https://github.com/Azure/azure-cli/blob/d9accde22cc493d7b4454baf2be8e3707c920920/src/azure-cli-core/azure/cli/core/commands/__init__.py#L1153-L1157

https://github.com/Azure/azure-cli/blob/dae19d0dfcad768f1dc19458f714334bacdf5a4f/src/azure-cli-core/azure/cli/core/util.py#L685-L690

Though I haven't got time to look into their actual usage.

Reply:
This line is enough to mitigate the issue. Moving the first request in the main thread will involve a huge change.
We cannot resolve them all because it's python issue https://github.com/python/cpython/issues/91351.

Reply:
Given https://github.com/Azure/azure-cli/issues/26297, do not use command index caching can walk around it. @jiasli

Reply:
When loading other modules, some of them will import `requests`. That's why it can mitigate the issue.